plone/app/linkintegrity/browser/info.py

```diff
@@ -67,7 +69,9 @@ def get_breaches(self, items=None):
                         "No object found for %s! Skipping", brain_to_delete
                     )
                     continue
-                for breach in self.get_breaches_for_item(obj):
```

Comment:
I'm not sure if I read the code correctly but shouldn't this check the breach of the `obj_to_delete` instead of the `obj` ?

Reply:
ouch ... I think there is something like `"depth": 1` missing in the `get_breaches_for_item` catalog query ... otherwise the recursive lookup does things multiple times for the same folder.

Reply:
@petschki I have rewritten a good chunk of the code, while maintaining backwards compatibility. In the Plone code base I just found usages for `get_breaches`. In case anyone uses in their 3rd party add-on the other methods, they should still be able to do so.